test/unit/org/apache/cassandra/metrics/CompactionMetricsTest.java

```diff
@@ -0,0 +1,229 @@
+    {
+
+        final long initialCompletedTasks = CompactionManager.instance.getCompletedTasks();
+        Assert.assertEquals(0L, CompactionManager.instance.getTotalCompactionsCompleted());
+        Assert.assertEquals(0L, CompactionManager.instance.getTotalBytesCompacted());
+
+        this.createTable("CREATE TABLE %s (pk int, ck int, a int, b int, PRIMARY KEY (pk, ck))");
+        this.getCurrentColumnFamilyStore().disableAutoCompaction();
+        for (int i = 0; i < 10; ++i)
+        {
+            this.execute("INSERT INTO %s (pk, ck, a, b) VALUES (" + i + ", 2, 3, 4)");
+        }
+
+        this.getCurrentColumnFamilyStore().forceBlockingFlush();
+        this.getCurrentColumnFamilyStore().forceMajorCompaction();
+
+        Assert.assertEquals(1L, CompactionManager.instance.getTotalCompactionsCompleted());
+        Assert.assertTrue(CompactionManager.instance.getTotalBytesCompacted() > 0L);
+        Assert.assertTrue(initialCompletedTasks < CompactionManager.instance.getCompletedTasks());
+    }
+
+    @Test
+    public void testCompactionMetricsForPendingTasks()
+    {
+        final ColumnFamilyStore cfs = MockSchema.newCFS();
+        final List<SSTableReader> sstables = this.createSSTables(cfs, 10, 0);
+        final Set<SSTableReader> toMarkCompacting = new HashSet<>(sstables.subList(0, 3));
+        final TestCompactionTask tct = new TestCompactionTask(cfs, toMarkCompacting);
+        final long initialPendingTasks = CompactionManager.instance.getMetrics().pendingTasks.getValue();
+        try
+        {
+            tct.start();
+            final List<CompactionInfo.Holder> activeCompactions = this.getActiveCompactionsForTable(cfs);
+            Assert.assertEquals(1L, activeCompactions.size());
+            Assert.assertEquals(activeCompactions.get(0).getCompactionInfo().getSSTables(), toMarkCompacting);
+            cfs.runWithCompactionsDisabled(() -> null, sstable -> !toMarkCompacting.contains(sstable), false, false, true);
+            Assert.assertEquals(1L, activeCompactions.size());
+            Assert.assertFalse(activeCompactions.get(0).isStopRequested());
+
+            // logger.info("testing data: {} {} {}", initialPendingTasks, CompactionManager.instance.getMetrics().pendingTasks.getValue(), CompactionManager.instance.getMetrics().pendingTasksByTableName.getValue().containsKey(cfs.keyspace.getName()));
+            Assert.assertTrue(CompactionManager.instance.getMetrics().pendingTasksByTableName.getValue().containsKey(cfs.keyspace.getName()));
+            Assert.assertTrue(CompactionManager.instance.getMetrics().pendingTasks.getValue() > initialPendingTasks);
+            logger.info("finish testing");
+        }
+        finally
+        {
+            tct.abort();
+        }
+    }
+
+    @Test
+    public void testCompactionMetricsForFailedTasks()
+    {
+
+        Assert.assertEquals(0, CompactionManager.instance.getMetrics().compactionsAborted.getCount());
```

Comment:
Nothing wrong functionally, but it might tighten things up a bit if we get a local reference to metrics and use that throughout.